src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/EcmaFormat/ReflectionTypeProvider.cs

```diff
@@ -0,0 +1,306 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
```

Comment:
(And the code that is generic is in the metadata reader, so the use is required if we continue to use those metadata parsers.